src/lib/io/sol-iio.c

```diff
@@ -654,6 +656,28 @@ channel_get_pure_name(const char *name)
 }
 
 static bool
+iio_set_oversampling_ratio(struct sol_iio_device *device, const struct sol_iio_config *config)
+{
+    char path[PATH_MAX];
+    struct sol_str_table *iter;
+
+    SOL_NULL_CHECK(device, false);
+
+    for (iter = config->oversampling_ratio_table; iter->key; iter++) {
+        if (!iter->val)
+            continue;
+
+        if (craft_filename_path(path, sizeof(path), CHANNEL_OVERSAMPLING_RATIO_DEVICE_PATH,
+            device->device_id, iter->key)) {
+            if (sol_util_write_file(path, "%d", iter->val) <= 0)
+                return false;
+        }
```

Comment:
`craft_filename_path` fails is not an error ? you are only returning false when `sol_util_write_file` fails.